gridstatus/ercot.py

```diff
@@ -432,6 +501,253 @@ def get_interconnection_queue(self, verbose=False):
+                unsupported = True
+        else:
+            unsupported = True
+
+        if unsupported:
+            raise NotSupported(
+                f"Market {market} and/or date {date} are not supported for ERCOT",
+            )
+
+    def _get_lmp_dam_latest(
+        self,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Gets today's data and filters all rows matching the maximum time"""
+        df = self._get_lmp_dam_today(
+            locations,
+            location_type,
+            verbose,
+        )
+        max_time = df["Time"].max()
+        df = df[df["Time"] == max_time]
+        return df
+
+    def _get_lmp_dam_today(
+        self,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Get day-ahead hourly Market LMP data for ERCOT"""
+        today_date = pd.Timestamp(pd.Timestamp.now(tz=self.default_timezone).date())
+        # adjust for DAM since it's published a day ahead
+        previous_date = today_date - pd.Timedelta("1D")
+        doc_url, publish_date = self._get_document(
+            report_type_id=DAM_SETTLEMENT_POINT_PRICES_RTID,
+            date=previous_date,
+            constructed_name_contains="csv.zip",
+            verbose=verbose,
+        )
+        if verbose:
+            print(f"Fetching {doc_url}", file=sys.stderr)
+        df = pd.read_csv(doc_url, compression="zip")
+
+        # fetch mapping
+        df["Market"] = Markets.DAY_AHEAD_HOURLY.value
+        df["Location Type"] = self._get_location_type_name(location_type)
+
+        mapping_df = self._get_settlement_point_mapping(verbose=verbose)
+        df = self._filter_by_location_type(df, mapping_df, location_type)
+        df = self._filter_by_locations(df, "SettlementPoint", locations)
+
+        df["Time"] = pd.to_datetime(
+            df["DeliveryDate"]
+            + " "
+            + (df["HourEnding"].str.split(":").str[0].astype(int) - 1)
+            .astype(str)
+            .str.zfill(2)
+            + ":00",
+        ).dt.tz_localize(self.default_timezone, ambiguous="infer")
+
+        df = df.rename(
+            columns={
+                "SettlementPointPrice": "LMP",
+                "SettlementPoint": "Location",
+            },
+        )
+
+        df = df[
+            [
+                "Location",
+                "Time",
+                "Market",
+                "Location Type",
+                "LMP",
+            ]
+        ]
+
+        df = df.reset_index(drop=True)
+        return df
+
+    def _get_lmp_rtm15_latest(
+        self,
+        locations: list = None,
+        location_type: str = None,
+        verbose=False,
+    ):
+        """Get Real-time 15-minute Market LMP data for ERCOT
+
+        https://www.ercot.com/mp/data-products/data-product-details?id=NP6-788-CD
+        """
+        today = pd.Timestamp(pd.Timestamp.now(tz=self.default_timezone).date())
```

Comment:
i believe this can be written using `normalize()` instead

```
>>> pd.Timestamp(pd.Timestamp.now().date())
Timestamp('2022-12-20 00:00:00')
>>> pd.Timestamp.now().normalize()
Timestamp('2022-12-20 00:00:00')
>>> pd.Timestamp.now().normalize() ==  pd.Timestamp(pd.Timestamp.now().date())
True
```

i'm guessing you were following existing code used elsewhere in this file, so if you could update those places too, that'd be very appreciated.

Reply:
I think we need to preserve `tz=self.default_timezone` because if this code is running in UTC but local time (US Central) is still in the previous day, then we want the local date, right?

So then this becomes:

```python
pd.Timestamp.now(tz=self.default_timezone).normalize()
```

Reply:
yep, that looks correct